cmd/control_loop/resources/resources.go

```diff
@@ -0,0 +1,136 @@
+package resources
+
+import (
+	"github.com/apigee/registry/rpc"
+	"github.com/golang/protobuf/ptypes"
+	"time"
+	"regexp"
+	"fmt"
+)
+
+type Resource interface {
+	Getartifact() string
+	Getspec() string
+	Getversion() string
+	Getapi() string
+	GetName() string
+	GetUpdateTimestamp() time.Time
+	ExtractResourceGroup (string) string
+}
+
+type SpecResource struct {
+	Spec *rpc.ApiSpec
+
+}
+
+func (s SpecResource) Getartifact() string {
```

Comment:
I think this and the following function names should use CamelCase (GetArtifact, etc).

Reply:
Done